ttnn/cpp/ttnn/operations/moreh/moreh_cumsum/moreh_cumsum.hpp

```diff
@@ -0,0 +1,37 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ttnn/decorators.hpp"
+#include "ttnn/operations/moreh/moreh_cumsum/device/moreh_cumsum_device_operation.hpp"
+
+namespace ttnn::operations::moreh::moreh_cumsum {
+struct MorehCumsum {
+    static Tensor invoke(
+        const Tensor& input,
+        const Tensor& output,
```

Comment:
I am confused. Is this a preallocated output? 
If so
* why is it const& here and why is it not optional here?
* why do you always create an output tensor here https://github.com/tenstorrent/tt-metal/pull/12296/files#diff-b4df615d6d6163c2ed02bef119ca5b210010279596c6f7841c41df470f29e0c8R63 ?

Reply:
I've modified the code to address your comment